src/excalidraw-app/data/index.ts

```diff
@@ -125,17 +125,27 @@ export const decryptAESGEM = async (
 };
 
 export const getCollaborationLinkData = (link: string) => {
-  if (link.length === 0) {
-    return;
-  }
   const hash = new URL(link).hash;
-  return hash.match(/^#room=([a-zA-Z0-9_-]+),([a-zA-Z0-9_-]+)$/);
+  const match = hash.match(/^#room=([a-zA-Z0-9_-]+),([a-zA-Z0-9_-]+)$/);
+  return match ? { roomId: match[1], roomKey: match[2] } : null;
+};
+
+export const generateCollaborationLinkData = async () => {
+  const roomId = await generateRandomID();
+  const roomKey = await generateEncryptionKey();
```

Comment:
Can you promise.all the two calls so that they don't need to wait on each other?

Reply:
Thought about it while I was refactoring this method, but it seemed like a micro-optim.

Just tested. It takes 0.4ms, and parallelizing it actually tends to increase it by 0.1ms at times, and makes it less readable :).